### ISSUE TITLE
Workflow Ordering Activities (Transitions)

### ISSUE THREAD

**Issue by ghost** [OrchardCMS/Orchard#8034]
I'm trying to get a work transitions in order . 

whenever i add a new activity its inserted as the first Activity in the transitions List from the workflowContext .

**Comment by sfmskywalker** [comment]
I'm not sure what your question is. Can you rephrase and or elaborate what you're trying to do, what you need to know, or how it's failing?

**Comment by ghost** [comment]
@sfmskywalker  Lets say we have a workflow which contains 10 activities each activity have only one outcome .

in the transitions record i want to get the first transition as index 0 and the seconde is 1 etc ...
because my current workflow contains 8 activities but each time i add an activity its added in the last index of the array `(Count - 1)` so i don't know which one is the starting point and which one will be trigged secondly using the transitions List in the `WorkflowContext`

so in short i just want to get the `activities ` on the `WorkflowDefinitionRecord ` sorted .

**Comment by sfmskywalker** [comment]
I'm not sure why you would want to do that, but you could just take the first activity (the one having zero incoming transitions), and then find the next activity by selecting the transition whose `SourceActivityId` is the one of the first activity. Having that transition, you now know the next activity which is stored in the transition's `DestinationActivityId` property. You can then continue walking this chain of activities, collecting them as you go and they will automatically be added to the list in the order they appear on the workflow.
E.g.

```csharp
var workflowType = await _workflowTypeStore.GetAsync("your workflow type ID");

// In case you want to get the start activity without depending on its `IsStart` property:
var destinationActivityIds = workflowType.Transitions.Select(x => x.DestinationActivityId);
var startActivity = workflowType.Activities.FirstOrDefault(x => !destinationActivityIds.Contains(x.ActivityId));

// Otherwise just use the IsStart property:
var startActivity = workflowType.Activities.FirstOrDefault(x => x.IsStart);

// Having a start activity, we can walk down the chain to collect connected activities:
var currentActivityId = startActivity.ActivityId;
var collectedTransitions = new List<TransitionRecord>();

while(currentActivityId != null)
{
   var transition = workflowType.Transitions.FirstOrDefault(x => x.SourceActivityId == currentActivityId);
 
   currentActivityId = transition?.DestinationActivityId;
   
   if(currentActivityId != null)
   {
      collectedTransitions.Add(transition);
   }
}
```

However, this code only works for workflows that only have one outbound transition, which makes it very brittle.

If you tell me what you're trying to achieve, then maybe I can suggest a better approach that would work for any type of workflow.

**Comment by ghost** [comment]
@sfmskywalker thank you for your reply . 

  what i'm trying to achieve is i'm making a dynamic Wizard using `Dynamic Forms` and `Workflows ` So i'm creating for example 3 Forms . 

and i'm using workflows with a custom activity , i'am redirecting the user from one page (Form) to another . 

so i want to be able to get how many Steps (Pages) i have .

this is easy however i need to sort those `Redirections`  on the workflow and then i would know that in workflow the Activity x (which contains a redirection to Page 1 (Step1) will be executed first and then the next redirection etc ...

**Comment by ghost** [comment]
using Transitions record is not sorted . because in the `index[0]` i'm getting a transitions which will be executed after some other transitions . so the transitions on the `WorkFlowContext ` class is not Sorted. 

the wizard works without this . but why i need it is to inform the customer what's the next step etc ..
